matchzoo/datapack.py

```diff
@@ -53,18 +53,36 @@ def __init__(self,
         :param left: Store the content or features for id_left.
         :param right: Store the content or features for
             id_right.
-        :param context: Hyper-parameter fitted during
-            pre-processing stage.
         """
         self._relation = relation
         self._left = left
         self._right = right
-        self._context = context
+
+    @property
+    def stage(self):
+        if 'label' in self._relation.columns:
+            return 'train'
+        else:
+            return 'predict'
```

Comment:
stage=`evaluate`

Reply:
`evaluate` requires `"label"` field as well right?

Reply:
@uduse if stage == evaluate, we do not fit parameters.

Reply:
@bwanglzu This isn't true anymore. I discussed this with Yixing yesterday and convinced him why abandoning all "stage" related stuff from user's side is a good thing. Now, `fit_transform` always fit and transform (well, it should).